Source/websocket/WebSocketLink.cpp

```diff
@@ -142,18 +136,22 @@ namespace Web {
                 if ((_controlStatus & REQUEST_CLOSE) != 0) {
                     dataFrame[result++] = FINISHING_FRAME | Protocol::CLOSE;
                     _controlStatus &= (~REQUEST_CLOSE);
-                    dataFrame[result++] = (_setFlags & MASKING_FRAME);
                 }
                 if (((_controlStatus & REQUEST_PING) != 0) && ((result + 1) < maxSendSize)) {
                     dataFrame[result++] = FINISHING_FRAME | Protocol::PING;
                     _controlStatus &= (~REQUEST_PING);
-                    dataFrame[result++] = (_setFlags & MASKING_FRAME);
                 }
                 if (((_controlStatus & REQUEST_PONG) != 0) && ((result + 1) < maxSendSize)) {
                     dataFrame[result++] = FINISHING_FRAME | Protocol::PONG;
                     _controlStatus &= (~REQUEST_PONG);
-                    dataFrame[result++] = (_setFlags & MASKING_FRAME);
                 }
+
+                dataFrame[result++] = (_setFlags & MASKING_FRAME);
```

Comment:
Let's bring this line back to where it was above. We need to set it unconditionally (see next comment), must for now be set to 0 when no masking used, and it will be easier when we later on (I created a Jira issue for it) need to handle the case that these frames can have application data.

Reply:
done